src/kokkos_ext/ArborX_DetailsKokkosExtUninitializedMemoryAlgorithms.hpp

```diff
@@ -0,0 +1,38 @@
+/****************************************************************************
+ * Copyright (c) 2024 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef ARBORX_DETAILS_KOKKOS_EXT_STD_MEMORY_HPP
+#define ARBORX_DETAILS_KOKKOS_EXT_STD_MEMORY_HPP
+
+#include <Kokkos_Assert.hpp>
+#include <Kokkos_Macros.hpp>
+
+#include <utility>
+
+namespace ArborX::Details::KokkosExt
+{
+
+template <class T, class... Args>
+KOKKOS_FUNCTION constexpr T *construct_at(T *p, Args &&...args)
+{
+  return ::new (const_cast<void *>(static_cast<void const volatile *>(p)))
+      T(std::forward<Args>(args)...);
```

Comment:
```suggestion
      T((Args&&)args...);
```
I would skip the std::forward and drop the include `<utility>`